frontend/src/component/PublicProfile/CustomFormManager.jsx

```diff
@@ -0,0 +1,311 @@
+  return (
+    <div className='DisplaySchemaPropertyString'>
+      <span
+        className={classnames(
+          'DisplaySchemaPropertyString__label',
+          { noLabel: !props.label }
+        )}
+      >
+        {props.label}
+      </span>
+      <span
+        className='DisplaySchemaPropertyString__value'
+        dangerouslySetInnerHTML={{ __html: props.value }}
+      />
+    </div>
+  )
+}
+
+const DisplaySchemaPropertyBoolean = props => {
+  return (
+    <div className='DisplaySchemaPropertyString'>
+      <span
+        className={classnames(
+          'DisplaySchemaPropertyString__label',
+          { noLabel: !props.label }
+        )}
+      >
+        {props.label}
+      </span>
+      <span className='DisplaySchemaPropertyString__value'>
+        <Checkbox
+          name={props.label}
+          checked={props.value}
+          onClickCheckbox={() => {}}
+          styleLabel={{ top: '8px', cursor: 'default' }}
+          styleCheck={{ top: '-5px' }}
+        />
+      </span>
+    </div>
+  )
+}
+
+const DisplaySchemaArray = props => {
+  if (!props.valueList) return null
+
+  if (props.valueList.every(value => typeof value === 'string')) {
+    return (
+      <DisplaySchemaPropertyString
+        label={props.label}
+        value={props.valueList.join(', ')}
+      />
+    )
+  }
+
+  return [
+    <DisplayTitle
+      label={props.label}
+      key={props.label}
+    />,
+    props.valueList.map((value, i) => {
+      if (!value) return null
+
+      const valueType = typeof value
+
+      if (valueType === 'object') {
+        return (
+          <div
+            className='DisplaySchemaArray'
+            key={`object_${props.currentKey}_${value.title}_${i}`}
+          >
+            <DisplaySchemaObject
+              schemaObject={props.schemaObject}
+              dataSchemaObject={value}
+              key={`object_${props.currentKey}_${value.title}__DisplaySchemaObject_${i}`}
+            />
+          </div>
+        )
+      }
+
+      return null
+    })
+  ]
+}
+
+const DisplaySchemaObject = props => {
+  if (!props.dataSchemaObject || !props.schemaObject) {
+    console.log('Error in DisplaySchemaObject, null props', props)
+    return null
+  }
+
+  const title = props.schemaObject.title
+    ? (
+      <DisplayTitle
+        label={props.schemaObject.title}
+        key={props.schemaObject.title}
+      />
+    )
+    : null
+
+  return [
+    title,
+    <div className='DisplaySchemaObject' key={`object_root_${props.nestedLevel}`}>
+      {Object.entries(props.dataSchemaObject).map(([key, value]) => {
+        const valueType = typeof value
+
+        if (Array.isArray(value)) {
+          return (
+            <DisplaySchemaArray
+              label={props.schemaObject.properties[key].title}
+              valueList={value}
+              currentKey={key}
+              schemaObject={props.schemaObject.properties[key].items}
+              key={`array_${key}`}
```

Comment:
```suggestion
              key={key}
```
key is unique across different types, no need to add a prefix.

You could have used switch / case here, but I'm fine with ifs. It's arguably more legible than switch / case anyway.

Reply:
I usually like switch cases but here I have two different tests, the Array.isArray() and the typeof so I kept everything in ifs

I generally agree with you about the prefix but I chose to keep it because every other renderer have keys with their types.

The key management was some of the most complex part of this development, having verbose keys helped me debugging. So I'd rather keep it